crypto/shamir.c

```diff
@@ -48,7 +48,7 @@ static void bitslice(uint32_t r[8], const uint8_t *x, size_t len) {
   for (arr_idx = 0; arr_idx < len; arr_idx++) {
     cur = (uint32_t)x[arr_idx];
     for (bit_idx = 0; bit_idx < 8; bit_idx++) {
-      r[bit_idx] |= ((cur & (1 << bit_idx)) >> bit_idx) << arr_idx;
+      r[arr_idx] |= ((cur >> bit_idx) & 1) << arr_idx;
```

Comment:
Is the change from `bit_idx ` to `arr_idx` intended?

Reply:
I just fixed as you commented.